Type inference failed. Expected type mismatch: inferred type is Boolean? but Boolean was expected

## 🐛 Bug Report

<!--- Summary description of the bug --->
kotlin_version = '1.2.30'

Launching lib/main.dev.dart on ONEPLUS A6000 in debug mode...
Initializing gradle...
Resolving dependencies...
Gradle task 'assembleDevDebug'...
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: /Users/sandeeepcmsm/.pub-cache/hosted/pub.dartlang.org/flutter_webview_plugin-0.2.1+2/android/src/main/java/com/flutter_webview_plugin/WebviewManager.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /Users/sandeeepcmsm/.pub-cache/hosted/pub.dartlang.org/flutter_webview_plugin-0.2.1+2/android/src/main/java/com/flutter_webview_plugin/WebviewManager.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
e: /Users/sandeeepcmsm/.pub-cache/hosted/pub.dartlang.org/google_api_availability-1.0.2/android/src/main/kotlin/com/baseflow/googleapiavailability/GoogleApiAvailabilityPlugin.kt: (29, 48): Type inference failed. Expected type mismatch: inferred type is Boolean? but Boolean was expected

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':google_api_availability:compileDebugKotlin'.
> Compilation error. See log for more details

### Reproduction steps

### Configuration
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, v0.9.7-pre.35, on Mac OS X 10.14 18A391, locale en-IN)
[✓] Android toolchain - develop for Android devices (Android SDK 28.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 10.0)
[✓] Android Studio (version 3.2)
[!] IntelliJ IDEA Community Edition (version 2018.1.5)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] VS Code (version 1.27.2)
[✓] Connected device (2 available)

! Doctor found issues in 1 category.
**Version:** 1.x

**Platform:** 
- [ ] :robot: Android

--- a/android/src/main/kotlin/com/baseflow/googleapiavailability/GoogleApiAvailabilityPlugin.kt
+++ b/android/src/main/kotlin/com/baseflow/googleapiavailability/GoogleApiAvailabilityPlugin.kt
@@ -26,11 +26,11 @@ class GoogleApiAvailabilityPlugin(val activity: Activity) : MethodCallHandler {
     override fun onMethodCall(call: MethodCall, result: Result) {
         when {
             call.method == "checkPlayServicesAvailability" -> {
-                val showDialog: Boolean = call.argument("showDialog")
+                val showDialog: Boolean? = call.argument("showDialog")
                 val googleApiAvailability = GoogleApiAvailability.getInstance()
                 val connectionResult = googleApiAvailability.isGooglePlayServicesAvailable(activity)
 
-                if (showDialog){
+                if (showDialog != null && showDialog){
                     googleApiAvailability.showErrorDialogFragment(activity, connectionResult, REQUEST_GOOGLE_PLAY_SERVICES)
                 }
 

--- a/example/android/build.gradle
+++ b/example/android/build.gradle
@@ -1,5 +1,5 @@
 buildscript {
-    ext.kotlin_version = '1.2.30'
+    ext.kotlin_version = '1.2.71'
     repositories {
         google()
         jcenter()

--- a/example/lib/main.dart
+++ b/example/lib/main.dart
@@ -18,16 +18,15 @@ class _MyAppState extends State<MyApp> {
   @override
   void initState() {
     super.initState();
-    //initPlatformState();
   }
 
   // Platform messages are asynchronous, so we initialize in an async method.
-  Future<void> initPlatformState([bool showDialog = false]) async {
+  Future<void> checkPlayServices([bool showDialog = false]) async {
     GooglePlayServicesAvailability playStoreAvailability;
     // Platform messages may fail, so we use a try/catch PlatformException.
     try {
-      playStoreAvailability =
-          await GoogleApiAvailability().checkGooglePlayServicesAvailability(showDialog);
+      playStoreAvailability = await GoogleApiAvailability()
+          .checkGooglePlayServicesAvailability(showDialog);
     } on PlatformException {
       playStoreAvailability = GooglePlayServicesAvailability.unknown;
     }
@@ -46,22 +45,27 @@ class _MyAppState extends State<MyApp> {
   Widget build(BuildContext context) {
     return new MaterialApp(
       home: new Scaffold(
-        appBar: new AppBar(
-          title: const Text('Plugin example app'),
-        ),
-        body: new ListView(children: <Widget>[
-            new MaterialButton(
-              onPressed: () => initPlatformState(), 
-              child: new Text("Get PlayServices availability"), 
-              color: Colors.red,
+          appBar: new AppBar(
+            title: const Text('Plugin example app'),
+          ),
+          body: new ListView(
+            children: <Widget>[
+              new MaterialButton(
+                onPressed: () => checkPlayServices(),
+                child: new Text("Get PlayServices availability"),
+                color: Colors.red,
               ),
-            new MaterialButton(
-              onPressed: () => initPlatformState(true), 
-              child: new Text("Get PlayServices availability with fix dialog"), color: Colors.redAccent,
+              new MaterialButton(
+                onPressed: () => checkPlayServices(true),
+                child:
+                    new Text("Get PlayServices availability with fix dialog"),
+                color: Colors.redAccent,
               ),
-            new Center(child: Text('Google Play Store status: ${_playStoreAvailability.toString().split('.').last}\n')),
-          ],
-        )),
+              new Center(
+                  child: Text(
+                      'Google Play Store status: ${_playStoreAvailability.toString().split('.').last}\n')),
+            ],
+          )),
     );
   }
 }

--- a/lib/google_api_availability.dart
+++ b/lib/google_api_availability.dart
@@ -13,8 +13,8 @@ part 'utils/codec.dart';
 class GoogleApiAvailability {
   factory GoogleApiAvailability() {
     if (_instance == null) {
-      final MethodChannel methodChannel =
-          const MethodChannel('flutter.baseflow.com/google_api_availability/methods');
+      final MethodChannel methodChannel = const MethodChannel(
+          'flutter.baseflow.com/google_api_availability/methods');
       _instance = new GoogleApiAvailability.private(methodChannel);
     }
     return _instance;
@@ -30,14 +30,14 @@ class GoogleApiAvailability {
   /// This feature is only available on Android devices. On any other platforms
   /// the [checkPlayServicesAvailability] method will always return
   /// [GooglePlayServicesAvailability.notAvailableOnPlatform].
-  Future<GooglePlayServicesAvailability>
-      checkGooglePlayServicesAvailability([bool showDialogIfNecessary = false]) async {
+  Future<GooglePlayServicesAvailability> checkGooglePlayServicesAvailability(
+      [bool showDialogIfNecessary = false]) async {
     if (!Platform.isAndroid) {
       return GooglePlayServicesAvailability.notAvailableOnPlatform;
     }
 
-    final dynamic availability =
-        await _methodChannel.invokeMethod('checkPlayServicesAvailability', {"showDialog":showDialogIfNecessary} );
+    final dynamic availability = await _methodChannel.invokeMethod(
+        'checkPlayServicesAvailability', {"showDialog": showDialogIfNecessary});
 
     return Codec.decodePlayServicesAvailability(availability);
   }

